UAlbertaBot/Maciej_Config.txt
{
    "Bot Info" :
    {
        "BotName" : "UAlbertaBot",
        "Authors" : "David Churchill",
        "PrintInfoOnStart" : false
    },
        
    "BWAPI" : 
    {
        "SetLocalSpeed"   : 0,
        "SetFrameSkip"    : 10,
        "EnableUserInput" : true,
        "EnableCompleteMapInformation" : true
    },
    
    "Micro" :
    {
        "UseSparcraftSimulation" : true,
    
        "WorkerDefense" : true,
        "WorkerDefensePerUnit" : 2,
        
        "InCombatRadius" : 1001,
        "RegroupRadius" : 301,
        "UnitNearEnemyRadius" : 601
    },
    
    "Macro" :
    {
        "BuildingSpacing" : 1,
        "PylonSpacing"   : 3
    },

    "Debug" :
    {
        "ErrorLogFilename"          : "d:/ualbertabot/UAlbertaBot/UAB_ErrorLog.txt",
        
        "PrintModuleTimeout"        : false,
        "DrawBuildOrderSearchInfo"  : false,
        "DrawUnitHealthBars"        : true,
        "DrawProductionInfo"        : true,
        "DrawEnemyUnitInfo"         : false,
        "DrawModuleTimers"          : false,
        "DrawResourceInfo"          : false,
        "DrawCombatSimInfo"         : false,
        "DrawUnitTargetInfo"        : false,
        "PrintBuildOrderSearchInfo" : false,
        "DrawBWTAInfo"              : false,
        "DrawMapGrid"               : false,
        "DrawSquadInfo"             : false,
        "DrawWorkerInfo"            : false,
        "DrawMouseCursorInfo"       : false,
        "DrawBuildingInfo"          : false,
        "DrawReservedBuildingTiles" : false,
        
        "AllDebugOff"               : false
    },
    
    "Modules" :
    {
        "UseGameCommander"      : true,
        "UseScoutManager"       : false,
        "UseCombatCommander"    : true,
        "UseBuildOrderSearch"   : true,
        "UseAutoObserver"       : false,
        "UseStrategyIO"         : true,
        "UseUnitCommandManager" : false
    },
    
    "Tools" :
    {
        "MapGridSize" : 320
    },
    
    "Strategy" :
    {
        "ProtossStrategyName"   : "Protoss_HTRush",
        "TerranStrategyName"    : "Terran_MarineRush",
        "ZergStrategyName"      : "Zerg_3HatchMuta",
        
        "ReadDirectory"         : "bwapi-data/write/",
        "WriteDirectory"        : "bwapi-data/write/",
        
        "Strategies" :
        {
            "Protoss_ZealotRush"    : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Gateway", "Gateway", "Probe", "Probe", "Zealot", "Pylon", "Zealot", "Zealot",  "Probe", "Zealot", "Zealot", "Probe", "Pylon", "Zealot", "Gateway", "Probe", "Pylon", "Probe", "Zealot", "Probe", "Zealot", "Zealot", "Zealot", "Zealot", "Pylon", "Probe", "Zealot", "Zealot", "Zealot" ]},
            "Protoss_DragoonRush"   : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Probe", "Gateway", "Probe", "Assimilator", "Probe", "Probe", "Cybernetics_Core", "Probe", "Probe", "Gateway", "Singularity_Charge", "Dragoon", "Gateway", "Pylon", "Dragoon", "Dragoon", "Probe", "Gateway", "Pylon", "Probe", "Dragoon", "Dragoon", "Dragoon" ]},
            "Protoss_HTRush"        : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Gateway", "Probe", "Assimilator", "Probe","Forge","Photon_Cannon" ,"Photon_Cannon" ,"Photon_Cannon","Cybernetics_Core", "Probe", "Citadel_of_Adun", "Probe", "Templar_Archives","Pylon","Psionic_Storm" ,"Photon_Cannon" ,"Photon_Cannon" ,"Photon_Cannon" ,  "High_Templar", "High_Templar", "Pylon", "High_Templar", "High_Templar", "Probe", "Pylon", "Probe" ]},
            "Protoss_DTRush"        : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Gateway", "Probe", "Assimilator", "Probe", "Cybernetics_Core", "Probe", "Citadel_of_Adun", "Probe", "Templar_Archives", "Gateway", "Dark_Templar", "Dark_Templar", "Pylon", "Dark_Templar", "Dark_Templar", "Probe", "Pylon", "Probe" ]},
	    "Terran_MarineRush"     : { "Race" : "Terran",  "OpeningBuildOrder" : ["SCV", "SCV", "SCV", "SCV", "SCV", "Supply_Depot", "SCV"]},
            "Zerg_ZerglingRush"     : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Overlord", "Spawning Pool"] },
            "Zerg_3HatchMuta"       : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Overlord", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Hatchery", "Zerg_Spawning_Pool", "Zerg_Drone", "Zerg_Drone", "Zerg_Hatchery", "Zerg_Extractor", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Overlord", "Zerg_Lair", "Zerg_Extractor", "Zerg_Drone", "Zerg_Drone", "Zerg_Zergling", "Zerg_Zergling", "Zerg_Zergling", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Overlord", "Zerg_Drone", "Zerg_Overlord", "Zerg_Drone", "Zerg_Overlord", "Zerg_Spire", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk"]}
        }
    }
}

UAlbertaBot/NotABot_Config.txt
{
    "Bot Info" :
    {
        "BotName" : "UAlbertaBot",
        "Authors" : "David Churchill",
        "PrintInfoOnStart" : false
    },
        
    "BWAPI" : 
    {
        "SetLocalSpeed"   : 6,
        "SetFrameSkip"    : 0,
        "EnableUserInput" : true,
        "EnableCompleteMapInformation" : false
    },
    
    "Micro" :
    {
        "UseSparcraftSimulation" : true,
    
        "WorkerDefense" : true,
        "WorkerDefensePerUnit" : 2,
        
        "InCombatRadius" : 1000,
        "RegroupRadius" : 300,
        "UnitNearEnemyRadius" : 600
    },
    
    "Macro" :
    {
        "BuildingSpacing" : 1,
        "PylonSpacing"   : 3
    },

    "Debug" :
    {
        "ErrorLogFilename"          : "d:/ualbertabot/UAlbertaBot/UAB_ErrorLog.txt",
        
        "PrintModuleTimeout"        : false,
        "DrawBuildOrderSearchInfo"  : false,
        "DrawUnitHealthBars"        : true,
        "DrawProductionInfo"        : true,
        "DrawEnemyUnitInfo"         : false,
        "DrawModuleTimers"          : false,
        "DrawResourceInfo"          : false,
        "DrawCombatSimInfo"         : false,
        "DrawUnitTargetInfo"        : false,
        "PrintBuildOrderSearchInfo" : false,
        "DrawBWTAInfo"              : false,
        "DrawMapGrid"               : false,
        "DrawSquadInfo"             : false,
        "DrawWorkerInfo"            : false,
        "DrawMouseCursorInfo"       : false,
        "DrawBuildingInfo"          : false,
        "DrawReservedBuildingTiles" : false,
        
        "AllDebugOff"               : false
    },
    
    "Modules" :
    {
        "UseGameCommander"      : true,
        "UseScoutManager"       : true,
        "UseCombatCommander"    : true,
        "UseBuildOrderSearch"   : true,
        "UseAutoObserver"       : true,
        "UseStrategyIO"         : true,
        "UseUnitCommandManager" : false
    },
    
    "Tools" :
    {
        "MapGridSize" : 320
    },
    
    "Strategy" :
    {
        "ProtossStrategyName"   : "NotABot",
        "TerranStrategyName"    : "Terran_MarineRush",
        "ZergStrategyName"      : "Zerg_ZerglingRush",
        
        "ReadDirectory"         : "bwapi-data/write/",
        "WriteDirectory"        : "bwapi-data/write/",
        
        "Strategies" :
        {
            "NotABot"               : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Probe", "Forge", "Probe", "Probe", "Probe", "Photon_Cannon", "Photon_Cannon", "Photon_Cannon", "Photon_Cannon", "Probe", "Pylon", "Gateway", "Assimilator", "Probe", "Cybernetics_Core"]},

            "Protoss_ZealotRush"    : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Gateway", "Gateway", "Probe", "Probe", "Zealot", "Pylon", "Zealot", "Zealot",  "Probe", "Zealot", "Zealot", "Probe", "Pylon", "Zealot", "Gateway", "Probe", "Pylon", "Probe", "Zealot", "Probe", "Zealot", "Zealot", "Zealot", "Zealot", "Pylon", "Probe", "Zealot", "Zealot", "Zealot" ]},
            "Protoss_DragoonRush"   : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Probe", "Gateway", "Probe", "Assimilator", "Probe", "Probe", "Cybernetics_Core", "Probe", "Probe", "Gateway", "Singularity_Charge", "Dragoon", "Gateway", "Pylon", "Dragoon", "Dragoon", "Probe", "Gateway", "Pylon", "Probe", "Dragoon", "Dragoon", "Dragoon" ]},
            "Protoss_DTRush"        : { "Race" : "Protoss", "OpeningBuildOrder" : ["Probe", "Probe", "Probe", "Probe", "Pylon", "Probe", "Gateway", "Probe", "Assimilator", "Probe", "Cybernetics_Core", "Probe", "Citadel_of_Adun", "Probe", "Templar_Archives", "Gateway", "Dark_Templar", "Dark_Templar", "Pylon", "Dark_Templar", "Dark_Templar", "Probe", "Pylon", "Probe" ]},
            "Terran_MarineRush"     : { "Race" : "Terran",  "OpeningBuildOrder" : ["SCV", "SCV", "SCV", "SCV", "SCV", "Supply_Depot", "SCV"]},
            "Zerg_ZerglingRush"     : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Overlord", "Spawning Pool"] },
            "Zerg_3HatchMuta"       : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Overlord", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Hatchery", "Zerg_Spawning_Pool", "Zerg_Drone", "Zerg_Drone", "Zerg_Hatchery", "Zerg_Extractor", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Overlord", "Zerg_Lair", "Zerg_Extractor", "Zerg_Drone", "Zerg_Drone", "Zerg_Zergling", "Zerg_Zergling", "Zerg_Zergling", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Overlord", "Zerg_Drone", "Zerg_Overlord", "Zerg_Drone", "Zerg_Overlord", "Zerg_Spire", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Drone", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk", "Zerg_Mutalisk"]}
        }
    }
}

UAlbertaBot/Source/Config.cpp
#include "Config.h"
#include "UABAssert.h"

namespace Config
{
    namespace Strategy
    {
        std::string ProtossStrategyName     = "Protoss_ZealotRush";
        std::string TerranStrategyName      = "Terran_MarineRush";
        std::string ZergStrategyName        = "Zerg_3HatchMuta";
        std::string StrategyName            = "Protoss_ZealotRush";
        std::string ReadDir                 = "bwapi-data/read/";
        std::string WriteDir                = "bwapi-data/write/";
    }

    namespace Modules							    
    {
        // the default tournament bot modules
        bool UsingGameCommander             = true;     // toggle GameCommander, effectively UAlbertaBot
        bool UsingScoutManager              = true;
        bool UsingCombatCommander           = true;
        bool UsingBuildOrderSearch          = true;     // toggle use of Build Order Search, currently no backup
        bool UsingAutoObserver              = false;
        bool UsingStrategyIO                = false;    // toggle the use of file io for strategy
        bool UsingUnitCommandManager        = false;    // handles all unit commands
		
        // extra things, don't enable unless you know what they are
        bool UsingBuildOrderDemo            = false;
    }

    namespace BotInfo
    {
        std::string BotName                 = "NotABot";
        std::string Authors                 = "Dave Churchill";
        bool PrintInfoOnStart               = false;
    }

    namespace BWAPIOptions
    {
        int SetLocalSpeed                   = 42;
        int SetFrameSkip                    = 0;
        bool EnableUserInput                = true;
        bool EnableCompleteMapInformation   = false;
    }
    
    namespace Tournament						
    {
        int GameEndFrame                    = 86400;	
    }
    
    namespace Debug								
    {
        bool PrintModuleTimeout             = true;	
        bool DrawBuildOrderSearchInfo       = true;
        bool DrawUnitHealthBars             = true;
        bool DrawProductionInfo             = true;
        bool DrawResourceInfo               = false;
        bool DrawWorkerInfo                 = false;
        bool DrawModuleTimers               = false;
        bool DrawReservedBuildingTiles      = false;
        bool DrawCombatSimulationInfo       = false;
        bool DrawBuildingInfo               = false;
        bool DrawMouseCursorInfo            = false;
        bool DrawEnemyUnitInfo              = false;
        bool DrawBWTAInfo                   = false;
        bool DrawMapGrid                    = false;
        bool DrawUnitTargetInfo             = false;
        bool DrawSquadInfo                  = false;
        bool AllDebugOff                    = false;

        std::string ErrorLogFilename        = "UAB_ErrorLog.txt";

        BWAPI::Color ColorLineTarget        = BWAPI::Colors::White;
        BWAPI::Color ColorLineMineral       = BWAPI::Colors::Cyan;
        BWAPI::Color ColorUnitNearEnemy     = BWAPI::Colors::Red;
        BWAPI::Color ColorUnitNotNearEnemy  = BWAPI::Colors::Green;
    }

    namespace Micro								
    {
        bool UseSparcraftSimulation         = true;
        bool WorkerDefense                  = true;     // whether or not we defend with workers when combat units die
        int WorkerDefensePerUnit            = 2;        // how many workers to assign to each unit attacking us
		
        int CombatRadius                    = 1000;     // radius of combat to consider units for Micro Search
        int CombatRegroupRadius             = 300;      // radius of units around frontmost unit we consider in regroup calculation
        int UnitNearEnemyRadius             = 600;      // radius to consider a unit 'near' to an enemy unit
    }

    namespace Macro
    {
        int BuildingSpacing                 = 1;
        int PylonSpacing                    = 3;
    }

    namespace Tools								
    {
        extern int MAP_GRID_SIZE            = 320;      // size of grid spacing in MapGrid
    }
}